soc/nordic/common/uicr/gen_uicr.py

```diff
@@ -192,11 +193,55 @@ def main() -> None:
         padding_len = periphconf_size - len(periphconf_combined)
         periphconf_final = periphconf_combined + bytes([0xFF for _ in range(padding_len)])
 
-        if kconfig.get("CONFIG_NRF_HALTIUM_UICR_PERIPHCONF") == "y":
+        # Configure PERIPHCONF field based on whether this is a recovery build
+        is_recovery_build = kconfig.get("CONFIG_NRF_HALTIUM_UICR_PERIPHCONF_RECOVERY") == "y"
+        is_normal_periphconf = kconfig.get("CONFIG_NRF_HALTIUM_UICR_PERIPHCONF") == "y"
+
+        if is_recovery_build:
+            # Recovery build: populate PERIPHCONF_RECOVERY field
+            uicr.PERIPHCONF_RECOVERY.ENABLE = ENABLED_VALUE
+            uicr.PERIPHCONF_RECOVERY.ADDRESS = periphconf_address
+            uicr.PERIPHCONF_RECOVERY.MAXCOUNT = math.floor(periphconf_size / 8)
+        elif is_normal_periphconf:
+            # Normal build: populate PERIPHCONF field
             uicr.PERIPHCONF.ENABLE = ENABLED_VALUE
             uicr.PERIPHCONF.ADDRESS = periphconf_address
             uicr.PERIPHCONF.MAXCOUNT = math.floor(periphconf_size / 8)
 
+        # Configure RECOVERY field if enabled
+        if kconfig.get("CONFIG_NRF_HALTIUM_UICR_RECOVERY") == "y":
+            uicr.RECOVERY.ENABLE = ENABLED_VALUE
+
+            # All RECOVERY field values must be present in config for configuration to proceed
+            processor_str = kconfig.get("CONFIG_NRF_HALTIUM_UICR_RECOVERY_PROCESSOR")
+            initsvtor_str = kconfig.get("CONFIG_NRF_HALTIUM_UICR_RECOVERY_INITSVTOR")
+            size4kb_str = kconfig.get("CONFIG_NRF_HALTIUM_UICR_RECOVERY_SIZE4KB")
+
+            if processor_str is None or initsvtor_str is None or size4kb_str is None:
+                raise ScriptError(
+                    "CONFIG_NRF_HALTIUM_UICR_RECOVERY is enabled but required RECOVERY "
+                    "configuration values are missing. Please ensure all of the following "
+                    "are configured: CONFIG_NRF_HALTIUM_UICR_RECOVERY_PROCESSOR, "
+                    "CONFIG_NRF_HALTIUM_UICR_RECOVERY_INITSVTOR, CONFIG_NRF_HALTIUM_UICR_RECOVERY_SIZE4KB"
+                )
+
+            # Set processor core (0=APPLICATION, 1=RADIOCORE)
+            processor = int(processor_str)
+            uicr.RECOVERY.PROCESSOR = processor
+
+            # Set INITSVTOR address (Vector Table Offset Register)
+            # Remove any quotes from the config value
+            initsvtor_str = initsvtor_str.strip('"')
+            if initsvtor_str.startswith("0x") or initsvtor_str.startswith("0X"):
+                initsvtor = int(initsvtor_str, 16)
+            else:
+                initsvtor = int(initsvtor_str, 10)
```

Comment:
We can use python's built in int parsing here:
```suggestion
            initsvtor = int(initsvtor_str, 0)
```

Reply:
fixed